openwhyd/openwhyd
# feat(security): add rate limiting to POST requests

What does this PR do / solve?
-----------------------------

Malicious users may want to brute force our `/login` endpoint to guess a user's credentials.

Overview of changes
-------------------

Rate limiting POST requests reduces the effectiveness of this kind of attack.

How to test this PR?
--------------------

After deploying this PR to production:

```sh
$ siege --concurrent=10 --reps=10 --benchmark --no-parser 'https://openwhyd.org/login' # should pass
$ siege --concurrent=10 --reps=10 --benchmark --no-parser 'https://openwhyd.org/login POST login=dummy' # should rate limit some of the requests
```
